pkg/executables/cmk.go

```diff
@@ -0,0 +1,363 @@
+package executables
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"gopkg.in/ini.v1"
+
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/templater"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+const (
+	cmkConfigFileName             = "cmk_tmp.ini"
+	cloudStackb64EncodedSecretKey = "CLOUDSTACK_B64ENCODED_SECRET"
+	cloudmonkeyInsecureKey        = "CLOUDMONKEY_INSECURE"
+)
+
+var (
+	//go:embed config/cmk.ini
+	cmkConfigTemplate string
+	requiredEnvsCmk   = []string{cloudStackb64EncodedSecretKey, cloudmonkeyInsecureKey}
+)
+
+const (
+	maxRetriesCmk    = 5
+	backOffPeriodCmk = 5 * time.Second
+)
+
+type Cmk struct {
+	writer     filewriter.FileWriter
+	executable Executable
+	retrier    *retrier.Retrier
+	execConfig *cmkExecConfig
```

Comment:
Why does this need to be stored in the struct? If it's transient, can we just create a new one on each command execution?

Reply:
Thanks for the review! This aligns with the pattern established by govc. Is that a sufficient reason? I'm not entirely sure what the alternative is here, when you suggest "create a new one on each command execution

Reply:
Sorry, my bad, I didn't explain it correctly.
I was only referring to `execConfig`

I can see a comment:
```go
// It must be cleaned after each execution to prevent side effects from past executions options
```

This makes me think it's transient, hence we don't need to store it in the `Cmk` struct and we can just build it ad-hoc for each command. Which it seems like we are doing it now with `setupExecConfig`. So I'm suggesting doing the same thing but without storing it in `Cmk`.

Reply:
Thanks for the clarification. I believe the comment may have been copied/pasted from the govc implementation unintentionally so I'll take a closer look at this.